### DISCUSSION TITLE
Counter in Textfield

### DISCUSSION THREAD

**Discussion by ingdesarr1** [flet-dev/flet#4224] [Q&A]
### Question

Good morning. I'm using a Textfield, and I want to limit the characters to 15, but I dont want to see the counter under the Textfield. I'm trying to delete using counter_text = None or '', but still appearing. In the documentation, show a property called counter, but it doesn't exist. Im using flet = 0.21.2. How can I delete this. Thanks
![image](https://github.com/user-attachments/assets/8c4ea44c-911f-4bc7-b29a-fe7399ddee7a)

### Code sample

```python
ft.TextField(
            label="Motivo",
            expand=2,
            visible=False,
            on_change=self.mensaje,
            max_length=15,
            counter_text='',
            input_filter=ft.TextOnlyInputFilter(),
            capitalization="CHARACTERS"
        )
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ap4499** [comment]
Hi - I think this is a bug, when the counter is set to a blank string or None, the documentation indicates that it should demonstrate the behaviour that you describe. 

I've tested this on 0.23.2.. I can test on the 0.25.dev later.. but I haven't seen this in the notes for it.

However, in the below - we get to the end result that you are after by using a space.. not perfect, but might work for you.



```
import flet as ft

def main(page):
    TF = ft.TextField(
                label="Motivo",
                expand=2,
                visible=True,
                # on_change=self.mensaje,
                max_length=15,
                input_filter=ft.TextOnlyInputFilter(),
                capitalization="CHARACTERS",
                counter_text = " "
            )


    page.add(TF)



if __name__ == "__main__":
    ft.app(main)
```

**Comment by ap4499** [comment]
> ### Question
> Good morning. I'm using a Textfield, and I want to limit the characters to 15, but I dont want to see the counter under the Textfield. I'm trying to delete using counter_text = None or '', but still appearing. In the documentation, show a property called counter, but it doesn't exist. Im using flet = 0.21.2. How can I delete this. Thanks ![image](https://private-user-images.githubusercontent.com/146734555/379850240-8c4ea44c-911f-4bc7-b29a-fe7399ddee7a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAzNjI4MzUsIm5iZiI6MTczMDM2MjUzNSwicGF0aCI6Ii8xNDY3MzQ1NTUvMzc5ODUwMjQwLThjNGVhNDRjLTkxMWYtNGJjNy1iMjlhLWZlNzM5OWRkZWU3YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDMxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAzMVQwODE1MzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zZGI5M2I3YzZjZDgxMDU4ODg1MmJmYjhjNDdlMDQyMzc1YTM1YzQ5NjQzYmYyOTg3ZWM3MWRhYjhjMjM5OTQzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.qi9fPU56CPwjT7oZOauKoaCO066ojsW90__C5h3BmCI)
> 
> ### Code sample
> ```python
> ft.TextField(
>             label="Motivo",
>             expand=2,
>             visible=False,
>             on_change=self.mensaje,
>             max_length=15,
>             counter_text='',
>             input_filter=ft.TextOnlyInputFilter(),
>             capitalization="CHARACTERS"
>         )
> ```
> 
> ### Error message
> _No response_
> 
> ### ------------------------------------------------------
> * [x]  I have searched for answers to my question both in the issues and in previous discussions.

Any luck with the above? Also, might be worth raising an issue.

**Reply by ingdesarr1** [reply]
If I put the space in counter_text, the counter dissapears, but the space of the counter still there, so, the Textfield dows not look aligned with the dropdown, and it looks bad.

**Reply by ap4499** [reply]
Yeah.. not ideal..

Whilst waiting for that bug fix, I think a temporary fix could be either:
- Adjust the alignment on that Row() to have them all at the top. At the moment, it looks like they are centre.
- Wrap the dropdown in a Container() that is of the same height as the TextField with counter

**Comment by ndonkoHenri** [comment]
I opened an issue to fix this: https://github.com/flet-dev/flet/issues/4273

**Comment by lastdayafter** [comment]
Try using the attribute value counter_style=TextStyle(size=0)
